packages/dds/tree/src/util/rangeMap.ts

```diff
@@ -147,73 +105,66 @@ export class RangeMap<T> {
-			if (entryLastKey < start) {
-				iBefore = i;
-			} else if (entry.start > end) {
-				iAfter = i;
-				break;
+		const lastDeletedKey = start + length - 1;
+		{
+			const entry = this.tree.getPairOrNextLower([start]);
+			if (entry !== undefined) {
+				const [key] = entry[0];
+				const { length: entryLength, value } = entry[1];
+				const lastEntryKey = key + entryLength - 1;
+				if (lastEntryKey >= start) {
+					// This entry overlaps with the deleted range, so we remove it.
+					this.tree.delete([key]);
+					if (key < start) {
+						// A portion of the entry comes before the delete range, so we reinsert that portion.
+						this.tree.set([key], { value, length: start - key });
+					}
+
+					if (lastEntryKey > lastDeletedKey) {
+						// A portion of the entry comes after the delete range, so we reinsert that portion.
+						this.tree.set([lastDeletedKey + 1], {
+							value,
+							length: lastEntryKey - lastDeletedKey,
+						});
+
+						return;
+					}
+				}
 			}
 		}
 
-		const numOverlappingEntries = iAfter - iBefore - 1;
-
-		if (numOverlappingEntries === 0) {
-			// No entry will be removed
-			return;
-		}
-
-		const iFirst = iBefore + 1;
-		const iLast = iAfter - 1;
-
-		for (let i = iFirst; i <= iLast; ++i) {
-			const entry = this.entries[i] ?? oob();
-			const entryLastKey = entry.start + entry.length - 1;
-			let isDirty = false;
-
-			if (entry.start >= start && entryLastKey <= end) {
-				// If the entry lies within the range to be deleted, remove it
-				this.entries.splice(i, 1);
-			} else {
-				// If the entry partially or completely overlaps with the range to be deleted
-				if (entry.start < start) {
-					// Update the endpoint and length of the portion before the range to be deleted
-					const lengthBefore = start - entry.start;
-					this.entries[i] = { ...entry, length: lengthBefore };
-					isDirty = true;
+		{
+			let entry = this.tree.nextHigherPair([start]);
+			while (entry !== undefined) {
+				const [key] = entry[0];
+				if (key > lastDeletedKey) {
+					return;
 				}
 
-				if (entryLastKey > end) {
-					// Update the startpoint and length of the portion after the range to be deleted
-					const newStart = end + 1;
-					const newLength = entryLastKey - end;
-					this.entries.splice(isDirty ? i + 1 : i, isDirty ? 0 : 1, {
-						start: newStart,
-						length: newLength,
-						value: entry.value,
+				const { length: entryLength, value } = entry[1];
+				const lastEntryKey = key + entryLength - 1;
+
+				this.tree.delete([key]);
+				if (lastEntryKey > lastDeletedKey) {
+					// A portion of the entry comes after the delete range, so we reinsert that portion.
+					this.tree.set([lastDeletedKey + 1], {
+						value,
+						length: lastEntryKey - lastDeletedKey,
 					});
+
+					return;
 				}
+
+				entry = this.tree.nextHigherPair([lastEntryKey]);
 			}
 		}
 	}
 }
 
 /**
  * Represents a contiguous range of values in the RangeMap.
```

Comment:
Did you remove the "This interface is used internally and should not be exposed to consumers." bit on purpose, or this is just a bad merge?

Reply:
It seemed like an unnecessary comment since the type wasn't exported.